### PR TITLE
handle edge case where key might expire in the middle of the multi exec for redis store

### PR DESCRIPTION
**Pull request by jmacpherson**
Discussion of the edge case: https://www.bennadel.com/blog/2785-volatile-keys-can-expire-mid-multi-transaction-in-redis-jedis.htm

Evidence of the ability to reprod with this library:
Screen Shot 2022-04-26 at 3 21 07 PM

### CODE REVIEW

**Comment by julienwilmet** [comment]
Does this also fix #39 and #44 ?

File: src/RedisStore.js [hunk 1]
```diff
@@ -58,7 +58,13 @@ class RedisStore extends Store {
 
       const seconds = Math.ceil(options.interval / 1000);
       await this.client.setEx(key, seconds.toString(), counter.toString());
-    }else {
+    } else if (dateEnd === -2 || dateEnd === -1) {
```
[the remark this answers is not in the document]
Comment (jmacpherson) [reply src/RedisStore.js:61]: -2 is indicative of a key value pair not existing. [Sauce](https://redis.io/commands/ttl/)

While I have not manually reproduced this case, it seems likely that it is possible in the same way that it is possible for the counter to exist at the time that it is queried, only to wind up not existing by the time we are setting a new value for the timer.

**Comment by jmacpherson** [comment]
> Does this also fix #39 and #44 ?

I can't speak to the `X-RateLimit-Reset` bug mentioned in #39, but it does address the ttl -1 issue, without resetting the counter back to 1 when doing so.

**Comment by panoti** [comment]
Hi @[jmacpherson](https://github.com/jmacpherson), if `dateEnd === -1 or -2` why you don't reset counter to `weight` (mean `1` as well)?

**Pull request merged**